error.go

```diff
@@ -69,6 +69,10 @@ var (
 	// ErrNotFound is the error returned to requests that don't match a registered handler.
 	ErrNotFound = NewErrorClass("not_found", 404)
 
+	// ErrMethodNotAllowed is the error returned to requests that have a registered handler
+	// but the used HTTP method does not match.
```

Comment:
This might be a bit clearer:
```go
// ErrMethodNotAllowed is the error returned to requests that match the path of a registered
// handler but not the HTTP method.
```